NgsildAgent/util/entrypoint.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+# Copyright (c) 2024 IB Systems GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+cd /app/util
+
+if [ "$DEBUG" = true ]; then
+    echo "Debugging is enabled"
+    echo "ARG1: $DEVICE_ID"
+    echo "ARG2: $GATEWAY_ID"
+    echo "ARG3: $KEYCLOAK_URL"
+    echo "ARG4: $REALM_ID"
+    echo "ARG5: $REALM_USER_PASSWORD"
+else
+    echo "Debugging is disabled"
+fi
+
+# Execute the first script with passed arguments
+./init-device.sh -k "$KEYCLOAK_URL" -r "$REALM_ID" "$DEVICE_ID" "$GATEWAY_ID"
```

Comment:
Does this mean that each time, the container is created it recreates the device? I don't think this is needed for K8s. It would be enough to mount the data folder to a pvc and then reuse the token/refresh token. The device is only recreated if the data folder is empty or recreation is forced by an ENV variable, e.g. FORCE_CREATION=true. Also, passing the user password into the container does create a security leak. The onboarding token should be passed in since if this is leaking it does not create such a high danger.